pkg/cortex/serve/poll/readiness.sh

```diff
@@ -16,7 +16,7 @@
 
 while true; do
     procs_ready="$(ls /mnt/workspace/proc-*-ready.txt 2>/dev/null | wc -l)"
-    if [ "$CORTEX_PROCESSES_PER_REPLICA" = "$procs_ready" ]; then
+    if [ "$CORTEX_PROCESSES_PER_REPLICA" = "$procs_ready" ] && curl --silent "localhost:${CORTEX_SERVING_PORT}/nginx_status" >/dev/null; then
```

Comment:
I think this is a good change, but may not be related, what do you think about separating it into a different PR?